idea/idea-frontend-independent/src/org/jetbrains/kotlin/idea/quickfix/AddWhenElseBranchFix.kt

```diff
@@ -49,9 +38,9 @@ class AddWhenElseBranchFix(element: KtWhenExpression) : KotlinQuickFixAction<KtW
         editor?.caretModel?.moveToOffset(endOffset + 1)
     }
 
-    companion object : KotlinSingleIntentionActionFactory() {
-        public override fun createAction(diagnostic: Diagnostic): AddWhenElseBranchFix? {
-            return diagnostic.psiElement.getNonStrictParentOfType<KtWhenExpression>()?.let(::AddWhenElseBranchFix)
+    companion object {
+        val factory: QuickFixesPsiBasedFactory<PsiElement> = quickFixesPsiBasedFactory { psiElement ->
```

Comment:
Can you put this directly on the companion object? Just to make it consistent with a few others (see`ReplaceWithDotCallFix`)

Reply:
Thanks! Fixed.